### PR TITLE
Add an option to dns_rfc2136 plugin to specify an authorative base domain.

### PR DESCRIPTION
**Pull request by avogler**
Added an option to dns_rfc2136 plugin to explicitly specify the base domain for which the DNS server is authorative. This solves problems with split horizon DNS configurations which do not correctly report authorative SOA records when queried without a TSIG key. Could be useful for other use cases.

### CODE REVIEW

File: certbot-dns-rfc2136/certbot_dns_rfc2136/dns_rfc2136.py [hunk 1]
```diff
@@ -171,23 +179,32 @@ def del_txt_record(self, record_name, record_content):
 
     def _find_domain(self, record_name):
         """
-        Find the closest domain with an SOA record for a given domain name.
+        Determine the authorative domain by either using the base domain specified in the
```
[the remark this answers is not in the document]
Comment (avogler) [reply certbot-dns-rfc2136/certbot_dns_rfc2136/dns_rfc2136.py]: Sure, I'll try to make it more clear.

**Comment by avogler** [comment]
@adferrand I changed the comments according to your remarks. Would be great if you could have a look at it.

**Comment by adferrand** [comment]
Yes of course, all good for me now!

**Comment by adferrand** [comment]
Oups forgot one thing, really sorry about that: could you make one other PR to add `certbot-dns-rfc2136` in the changelog, in the list starting L25? This informs our integrator that the plugin has changed and should be repackaged.

I will merge your new PR right after.

**Comment by avogler** [comment]
Yes sure, no problem.

**Pull request merged**